slides/segwit-transaction-malleability.tex

```diff
@@ -0,0 +1,351 @@
+\end{scriptsize}
+\vspace{1em}
+The entire transaction data \alert{no longer} hashes to:\\
+\begin{scriptsize}
+	\texttt{f4184fc596403b9d638783cf57adfe4c75c605f6356fbc91338530e9831e9e16}
+\end{scriptsize}
+\end{frame}
+%%%
+
+%%%
+\begin{frame}{Segregated Witness (SegWit) Transactions}
+	\begin{figure}
+		\input{../assets/figures/transaction-structure-segwit.tex}	
+	\end{figure}
+	\vspace{1em}
+	
+	\uncover<3->{$\rightarrow$ The \texttt{scriptSig} is always the same. The transaction's hash value is now an unambiguous transaction identifier.}
+\end{frame}
+%%%
+
+%%%
+\begin{frame}{Sort Fork Implementation}
+		From the \href{https://cryptolectures.teachable.com/courses/bitcoin-blockchain-and-cryptoassets/lectures/30728731}{\link Fork Theory lecture}:\\
+	\begin{figure}
+		\input{../assets/figures/soft-fork-segwit.tex}	
+	\end{figure}
+	\vspace{1em}
+	\begin{scriptsize}
+	\uncover<2->{
+		\textbf{If $r_{new} > r_{old}$:}\\
+		\begin{itemize}
+			\item \textbf{Nodes following $S_{old}$:} see SegWit transactions as \texttt{ANYONECANSPEND} and accept blocks with SegWit transactions
+			\item \textbf{Nodes following $S_{new}$:} if \texttt{scriptSig} is empty $\rightarrow$ check \texttt{witness} for valid signature.
+			\item The stricter rule set results in one dominant chain.
+		\end{itemize}
+	}
+	\vspace{1em}
+	\uncover<3->{
+		\textbf{If $r_{new} < r_{old}$:}\\
+		\begin{itemize}
+			\item \textbf{Nodes following $S_{old}$:} consensus relevant nodes (CRNs) create new blocks where they spend the \texttt{ANYONECANSPEND} outputs themselves.
+			\item \textbf{Nodes following $S_{new}$:} check \texttt{witness} for valid signature and reject blocks where CRNs spend SegWit transactions themselves.
+			\item Consensus conflict! Who can spend the output?
+		\end{itemize}
+	}
+	\end{scriptsize}
+\end{frame}
+%%%
+
+%%%
+\begin{frame}{SegWit Transaction Types (P2WPKH)}
+	Pay-to-witness-public-key-hash (P2WPKH) is the SegWit equivalent to P2PKH.\\
+	\vspace{1.5em}
+	
+	\shadowbox{
+		\begin{minipage}[c]{4.1in}
+			\texttt{scriptSig:}\\
+			\texttt{scriptPubKey:\ 0 <pubKeyHash>}\\
+			\texttt{witness:\ <sig> <pubKey>}
+		\end{minipage}
+	}
+	\vspace{1.5em}
+
+	\textbf{Non-SegWit nodes:} output can be spent without the need for a signature.\\
+	\vspace{0.5em}
+	\textbf{SegWit nodes:} ''look for and verify the signature in the witness data."
+\end{frame}
+%%%
+
+%%%
+\begin{frame}{SegWit Transaction Types (P2WSH)}
+	Pay-to-witness-script-has (P2WSH) is the SegWit equivalent to P2SH.\\
+	\vspace{1.5em}
+	
+	\shadowbox{
+		\begin{minipage}[c]{4.1in}
+			\texttt{scriptSig:}\\
+			\texttt{scriptPubKey:\ 0 <scriptHash256>}\\
+			\texttt{witness:\ Any valid script}
+		\end{minipage}
+	}
+	\vspace{1.5em}
+
+	\textbf{Non-SegWit nodes:} output can be spent by anyone.\\
+	\vspace{0.5em}
+	\textbf{SegWit nodes:} ''look for and verify the script in the witness data."
+\end{frame}
+%%%
+
+%%%
+\begin{frame}{P2SH Embedded SegWit Transactions}
```

Comment:
This slide is not that intuitive, but probably will be clear if presented in the video.